app/controllers/chat_channels_controller.rb

```diff
@@ -240,6 +241,25 @@ def retry_archive
     render json: success_json
   end
 
+  def toggle_open_status
+    params.require(:chat_channel_id)
+
+    chat_channel = ChatChannel.find_by(id: params[:chat_channel_id])
+    raise Discourse::NotFound if chat_channel.blank?
+
+    if chat_channel.open?
+      guardian.ensure_can_change_channel_status!(chat_channel, :closed)
+      chat_channel.close!(current_user)
+    elsif chat_channel.closed?
+      guardian.ensure_can_change_channel_status!(chat_channel, :open)
+      chat_channel.open!(current_user)
+    else
+      raise Discourse::InvalidAccess
```

Comment:
Is it possible for a channel to have any other state?

Reply:
O nevermind this :) I just saw the specs.

Reply:
Yes, archived and read_only, those are only done via the `archive` endpoint though. I will change to dedicated endpoints as your other comment suggests.